internal/resourcemapping/resourcemapping.go

```diff
@@ -0,0 +1,186 @@
+	// resource for a given monitored resource type.
+	monitoredResourceMappings = map[string]map[string]struct {
+		// OTel resource keys to try and populate the resource label from. For entries with
+		// multiple OTel resource keys, the keys' values will be coalesced in order until there
+		// is a non-empty value.
+		otelKeys []string
+		// If none of the otelKeys are present in the Resource, fallback to this literal value
+		fallbackLiteral string
+	}{
+		gceInstance: {
+			zone:       {otelKeys: []string{semconv.AttributeCloudAvailabilityZone}},
+			instanceID: {otelKeys: []string{semconv.AttributeHostID}},
+		},
+		k8sContainer: {
+			location:      {otelKeys: []string{semconv.AttributeCloudAvailabilityZone}},
+			clusterName:   {otelKeys: []string{semconv.AttributeK8SClusterName}},
+			namespaceName: {otelKeys: []string{semconv.AttributeK8SNamespaceName}},
+			podName:       {otelKeys: []string{semconv.AttributeK8SPodName}},
+			containerName: {otelKeys: []string{semconv.AttributeK8SContainerName}},
+		},
+		k8sPod: {
+			location:      {otelKeys: []string{semconv.AttributeCloudAvailabilityZone}},
+			clusterName:   {otelKeys: []string{semconv.AttributeK8SClusterName}},
+			namespaceName: {otelKeys: []string{semconv.AttributeK8SNamespaceName}},
+			podName:       {otelKeys: []string{semconv.AttributeK8SPodName}},
+		},
+		k8sNode: {
+			location:    {otelKeys: []string{semconv.AttributeCloudAvailabilityZone}},
+			clusterName: {otelKeys: []string{semconv.AttributeK8SClusterName}},
+			nodeName:    {otelKeys: []string{semconv.AttributeK8SNodeName}},
+		},
+		k8sCluster: {
+			location:    {otelKeys: []string{semconv.AttributeCloudAvailabilityZone}},
+			clusterName: {otelKeys: []string{semconv.AttributeK8SClusterName}},
+		},
+		awsEc2Instance: {
+			instanceID: {otelKeys: []string{semconv.AttributeHostID}},
+			region:     {otelKeys: []string{semconv.AttributeCloudAvailabilityZone}},
+			awsAccount: {otelKeys: []string{semconv.AttributeCloudAccountID}},
+		},
+		genericTask: {
+			location: {
+				otelKeys: []string{
+					semconv.AttributeCloudAvailabilityZone,
+					semconv.AttributeCloudRegion,
+				},
+				fallbackLiteral: "global",
+			},
+			namespace: {otelKeys: []string{semconv.AttributeServiceNamespace}},
+			job:       {otelKeys: []string{semconv.AttributeServiceName}},
+			taskID:    {otelKeys: []string{semconv.AttributeServiceInstanceID}},
+		},
+		genericNode: {
+			location: {
+				otelKeys: []string{
+					semconv.AttributeCloudAvailabilityZone,
+					semconv.AttributeCloudRegion,
+				},
+				fallbackLiteral: "global",
+			},
+			namespace: {otelKeys: []string{semconv.AttributeServiceNamespace}},
+			nodeID:    {otelKeys: []string{semconv.AttributeHostID, semconv.AttributeHostName}},
+		},
+	}
+)
+
+type GceResource struct {
+	Type   string
+	Labels map[string]string
+}
+
+type ReadOnlyAttributes interface {
+	GetString(string) (string, bool)
+}
+
+// Converts from a set of OTEL resource attributes into a
```

Comment:
```suggestion
// ResourceAttributesToMonitoredResource converts from a set of OTEL resource attributes into a
```
Function comments should start with the name of the function.